swap_meet/vendor.py

```diff
@@ -1,2 +1,229 @@
+  - It removes the first item from the friend's `inventory`, and adds the instances first item
+  - It returns `True`
+  - If either itself or the friend have an empty `inventory`, the method returns `False`
+
+### Wave 6
+
+In Wave 6 we will write two methods, `get_best_by_category` and `swap_best_by_category`.
+
+- `Vendor`s have a method named `get_best_by_category`, which will get the item with the best condition in a certain category
+  - It takes one argument: a string that represents a category
+  - This method looks through the instance's `inventory` for the item with the highest `condition` and matching `category`
+    - It returns this item
+    - If there are no items in the `inventory` that match the category, it returns `None`
+    - It returns a single item even if there are duplicates (two or more of the same item with the same condition)
+
+The remaining tests in wave 6 imply:
+
+- `Vendor`s have a method named `swap_best_by_category`, which will swap the best item of certain categories with another `Vendor`
+  - It takes in three arguments
+    - `other`, which represents another `Vendor` instance to trade with
+    - `my_priority`, which represents a category that the `Vendor` wants to receive
+    - `their_priority`, which represents a category that `other` wants to receive
+  - The best item in my inventory that matches `their_priority` category is swapped with the best item in `other`'s inventory that matches `my_priority`
+    - It returns `True`
+    - If the `Vendor` has no item that matches `their_priority` category, swapping does not happen, and it returns `False`
+    - If `other` has no item that matches `my_priority` category, swapping does not happen, and it returns `False`
+
+"""
+
+from asyncio.format_helpers import _format_callback_source
+
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    #with .pop()
+    def remove(self, item):
+        for i in range(len(self.inventory)):
+            if self.inventory[i] == item:
+                removed = self.inventory.pop(i)
+                return removed
+        return False
+
+    #with .remove()
+    # def remove(self,item):
+    #     aList = self.inventory
+    #     for thing in aList:
+    #         if thing == item:
+    #             self.inventory.remove(thing)
+    #             return thing
+    #     return False
+
+    def get_by_category(self, category):
+        itemList = []
+        
+        for item in self.inventory:
+            if item.category == category:
+                itemList.append(item)
+        #returns a list of `Item`s in the inventory with that category
+        return itemList
```

Comment:
👍  Nice work!

Reply:
Don't forget to use snake case for variables in Python (as according to the pep8 guidelines https://peps.python.org/pep-0008/#descriptive-naming-styles).

So in this case, `itemList` should be `item_list` instead. In Javascript, we'll switch over to camelcase for names!

Reply:
We can also write this function in one line using list comprehension:

```python
    def get_by_category(self, category):
        return [item for item in self.inventory if item.category == category]
```